autoload/go/jobcontrol.vim

```diff
@@ -93,7 +93,7 @@ endfunction
 " it'll be closed.
 function! s:on_exit(job_id, data)
   let std_combined = self.stderr + self.stdout
-  if empty(std_combined)
+  if jobwait([a:job_id])[0] > 0
```

Comment:
This is a highly critical path as it's used by many applications. As I know relying on exit code might produce wrong results. I don't remember which subcommand or command, but there was a case where I wanted to get the output regardless of job id. Do we need to change it here ?